### PR TITLE
ref(onboarding-wizard): Improve interface to wizard

### PR DESCRIPTION
**Pull request by evanpurkhiser**
- The types for OnboardingTasks have been split into OnboardingTaskDescriptor and OnboardingTaskStatus. OnboardingTask is now an intersection of the two types

 - `featureLocation` has been renamed to `actionType` and now accepts the values [`action`, `app`, and `external`].

   `app` replaces project and organization and now uses the navigation.nanavigateTo action to determine the needed slugs using the ContextPickerModal.

   `external` replaces `absolute` and is simply a more consistent name (as this now also uses `ExternalLink`)

   `action` replaces `modal` and allows a `action` option to be specified on the task, `location` is now always type `string`.

 - The ProjectChooser view has been completely removed since this is now replaced by the context picker modal.

 - TodoItem and TodoList have both been converted to typescript

 - TodoItem has lost it's ability to 'expand' items, as they are now always visually expanded.

 - TodoItem has had some visual stylistic improvements made

   * The entire item is now clickable
   * The text is now variants of gray
   * When a task is complete, the checkmark circle is now also green

image

### CODE REVIEW

File: src/sentry/static/sentry/app/components/onboardingWizard/toDoItem.tsx [hunk 1]
```diff
@@ -261,9 +243,16 @@ const ItemHeader = styled('h4')`
...
 `;
 
-const SkipButton = styled(Button)`
+const SkipButton = styled(Button)<{hide: boolean}>`
   display: ${p => p.hide && 'none'};
   margin-top: ${space(1.5)};
 `;
 
-export default withOrganization(TodoItem);
+// XXX(epurkhiser): The withRouter HoC has incorrect typings. It does not
```
Comment (jusigler) [inline src/sentry/static/sentry/app/components/onboardingWizard/toDoItem.tsx]: HoC?

**Review by jusigler — COMMENTED** [review]
Are we going to replace the deleted unit tests?

File: src/sentry/static/sentry/app/components/onboardingWizard/toDoItem.tsx (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/onboardingWizard/toDoItem.tsx]: Higher order component

**Comment by EvanPurkhiser** [comment]
Nothing to replace. We deleted project chooser :)

File: src/sentry/static/sentry/app/components/onboardingWizard/getOnboardingTasks.tsx [hunk 2]
```diff
@@ -32,8 +34,8 @@ export default function getOnboardingTasks(organization: Organization): Onboardi
       detailedDescription: t('Choose your platform and send an event.'),
       skippable: false,
       prereq: [1],
-      featureLocation: 'project',
-      location: 'settings/install/',
+      actionType: 'app',
+      location: `/settings/${organization.slug}/projects/:projectId/install/`,
```
Comment (maheskett) [inline src/sentry/static/sentry/app/components/onboardingWizard/getOnboardingTasks.tsx]: What should happen when the first step (creating a project) hasn't been completed yet? Previously, the user was able to create a project from this location as well.

File: src/sentry/static/sentry/app/components/onboardingWizard/getOnboardingTasks.tsx [hunk 3]
```diff
@@ -85,8 +87,8 @@ export default function getOnboardingTasks(organization: Organization): Onboardi
       ),
       skippable: true,
       prereq: [1, 2],
-      featureLocation: 'project',
-      location: 'settings/release-tracking/',
+      actionType: 'app',
+      location: `/settings/${organization.slug}/projects/:projectId/release-tracking/`,
```
Comment (maheskett) [inline src/sentry/static/sentry/app/components/onboardingWizard/getOnboardingTasks.tsx]: Same as above. What should happen when the first step (creating a project) hasn't been completed yet?

File: src/sentry/static/sentry/app/components/onboardingWizard/toDoItem.tsx [hunk 4]
```diff
@@ -123,39 +102,39 @@ class TodoItem extends React.Component {
...
-          <Description>{description}</Description>
-          <SkipButton
-            size="xsmall"
-            hide={!showSkipButton}
-            onClick={this.toggleConfirmation}
-          >
-            {t('Skip task')}
-          </SkipButton>
+            <Description>{description}</Description>
+            <SkipButton
```
Comment (maheskett) [inline src/sentry/static/sentry/app/components/onboardingWizard/toDoItem.tsx]: Are you able to skip with `SkipButton` inside `Action`?

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/onboardingWizard/toDoItem.tsx]: Yeah, made sure that works. Does some `preventDefault` magic.

File: src/sentry/static/sentry/app/components/onboardingWizard/getOnboardingTasks.tsx (hunk 3 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/onboardingWizard/getOnboardingTasks.tsx]: Ahh, we should rethink these to be disabled in that case. I'm okay just having the selector be empty in that case.

File: src/sentry/static/sentry/app/components/onboardingWizard/toDoItem.tsx [hunk 5]
```diff
@@ -1,108 +1,87 @@
...
 
-  recordAnalytics(action) {
+  recordAnalytics(action: string) {
     const {organization, task} = this.props;
     trackAnalyticsEvent({
       eventKey: 'onboarding.wizard_clicked',
       eventName: 'Onboarding Wizard Clicked',
       organization_id: organization.id,
-      todo_id: parseInt(task.task, 10),
+      todo_id: task.task,
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/onboardingWizard/toDoItem.tsx]: Should this be int? https://github.com/getsentry/reload/blob/master/reload_app/events.py#L562

**Review by billyvg — APPROVED** [review]
rip projectChooser

**Comment by billyvg** [comment]
o.o https://travis-ci.org/getsentry/sentry/jobs/650673134#L1377

File: src/sentry/static/sentry/app/components/onboardingWizard/toDoItem.tsx (hunk 5 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/onboardingWizard/toDoItem.tsx]: It already is an int

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/onboardingWizard/toDoItem.tsx]: typescript baby

**Pull request merged**